### PR TITLE
feat: added handling of prisma validation errors to workflow controller

### PR DESCRIPTION
**Pull request by chesterkmr**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 [Issue](https://github.com/ballerine-io/ballerine/issues/493)

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: services/workflows-service/src/common/filters/PrismaClientValidation.filter.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    // Exception message of this error could not be sent to client for security reasons.
+    // Message exposes internals of application which end-user should not know about.
+    // Writting details to logs which could be parsed in future.
+    this.logger.error(`Prisma validation failed.\n ${exception.message}`);
+
+    response.status(HttpStatus.BAD_REQUEST);
+
+    response.json({
+      statusCode: HttpStatus.BAD_REQUEST,
+      message: 'Validation failed.Details written to logs.',
```
Comment (alonp99) [inline services/workflows-service/src/common/filters/PrismaClientValidation.filter.ts:23]: Thanks @chesterkmr , is there a way to return more informative error for the user? about which field caused the issue for example

File: services/workflows-service/src/workflow/workflow.controller.external.ts [hunk 2]
```diff
@@ -114,6 +115,7 @@ export class WorkflowControllerExternal {
   @UseKeyAuthGuard()
   @common.HttpCode(200)
   @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
+  @UseFilters(new PrismaClientValidationFilter())
```
Comment (alonp99) [inline services/workflows-service/src/workflow/workflow.controller.external.ts:118]: Can you apply this filter globally? when thinking about it, must of the endpoint interacts with the db

File: services/workflows-service/src/common/filters/PrismaClientValidation.filter.ts (hunk 1 above)
Comment (chesterkmr) [reply services/workflows-service/src/common/filters/PrismaClientValidation.filter.ts:23]: @alonp99 Unfortunately its not possible to return more details of error to user.According to Prisma docs `PrismaClientValidationError` provides only human readable `message` which is exposes internal information about application which not supposed to get outside of application to end users.

File: services/workflows-service/src/workflow/workflow.controller.external.ts (hunk 2 above)
Comment (chesterkmr) [reply services/workflows-service/src/workflow/workflow.controller.external.ts]: Its set

File: services/workflows-service/src/common/filters/PrismaClientValidation.filter.ts (hunk 1 above)
Comment (chesterkmr) [reply services/workflows-service/src/common/filters/PrismaClientValidation.filter.ts]: @alonp99 Its done,please take a look

File: services/workflows-service/src/common/filters/prisma-client-validation-filter/utils/PrismaValidationExceptionParser.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * @constructor
+ * @param {Prisma.PrismaClientValidationError} exception - Prisma validation exception
+ * @description Parsing exception for errors and convert them to JSON format
+ * Excluding referernces to prisma and application internals
+ * @method parse @returns {PrismaValidationExceptionParseResult} record of entries where key is paramName and its value errorReason
+ */
+
+export class PrismaValidationExceptionParser {
+  // Looking for matches in following string format: Argument {{paramName}}: Got {{errorReason}} on prisma
+  private pattern = new RegExp(/Argument ([^:]+):[^]+?Got ([^]+?) on prisma/, 'g');
```
Comment (alonp99) [inline services/workflows-service/src/common/filters/prisma-client-validation-filter/utils/PrismaValidationExceptionParser.ts:15]: I checked a common issue - an unknown argument and got empty response:
```
{
    "statusCode": 400,
    "message": "Validation failed.",
    "result": {}
}
```

the error string is: 

```
Unknown arg `companyName1` in data.companyName1 for type BusinessCreateInput. Did you mean `companyName`? Available args:
```

Comment (alonp99) [reply services/workflows-service/src/common/filters/prisma-client-validation-filter/utils/PrismaValidationExceptionParser.ts:15]: If you need the curl i made, ping me on slack

File: services/workflows-service/src/common/filters/prisma-client-validation-filter/utils/PrismaValidationExceptionParser.test.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { PrismaValidationExceptionParser } from './PrismaValidationExceptionParser';
+import { Prisma } from '@prisma/client';
+
+describe('PrismaValidationExceptionParser', () => {
```
Comment (alonp99) [inline services/workflows-service/src/common/filters/prisma-client-validation-filter/utils/PrismaValidationExceptionParser.test.ts:4]: Good job

File: services/workflows-service/src/common/filters/prisma-client-validation-filter/utils/PrismaValidationExceptionParser.ts (hunk 3 above)
Comment (chesterkmr) [reply services/workflows-service/src/common/filters/prisma-client-validation-filter/utils/PrismaValidationExceptionParser.ts]: @alonp99 Added handler for this kind of errors

**Pull request merged**